src/rgw/rgw_cache_driver.h

```diff
@@ -24,7 +25,7 @@ typedef std::function<void(const DoutPrefixProvider* dpp, std::string& key, std:
 			    const rgw_obj_key& obj_key, optional_yield y)> ObjectDataCallback;
 
 typedef std::function<void(const DoutPrefixProvider* dpp, std::string& key, uint64_t offset, uint64_t len, std::string version,
-        bool dirty, optional_yield y, std::string& restore_val)> BlockDataCallback;
+        bool dirty, optional_yield y, std::vector<std::string>& restore_vals)> BlockDataCallback;
```

Comment:
In future we may have other attrs to restore (like the dirty flag), it will be better to use a map.